include/zephyr/drivers/display.h

```diff
@@ -247,6 +247,10 @@ static inline int display_write(const struct device *dev, const uint16_t x,
 	struct display_driver_api *api =
 		(struct display_driver_api *)dev->api;
 
+	if (api->write == NULL) {
+		return -ENOTSUP;
```

Comment:
Someone tried this before, but somehow it got stuck. I am not sure what is right, ENOTSUP or ENOSYS, https://docs.zephyrproject.org/latest/develop/api/design_guidelines.html is not clear at all. e.g. https://github.com/zephyrproject-rtos/zephyr/blob/673f38f3d1ccbb8d66edce195805deae439272d4/include/zephyr/drivers/auxdisplay.h#L327-L329 uses ENOSYS.
cc @nordicjm @kartben

Reply:
`ENOSYS` is correct here, from that page:
```
- APIs that are supported but not implemented shall return ``-ENOSYS``.

- Optional APIs that are not supported by the hardware should be implemented and
  the return code in this case shall be ``-ENOTSUP``.

- When an API is implemented, but the particular combination of options
  requested in the call cannot be satisfied by the implementation the call shall
  return ``-ENOTSUP``. (For example, a request for a level-triggered GPIO interrupt on
  hardware that supports only edge-triggered interrupts)
```

This is the top one. The second one applies in a driver for a specific peripheral e.g. display driver for hn238 but not the overall display.h file

Reply:
I'll change to -ENOSYS.